### PR TITLE
[FIX] Receiving notifications from removed server

### LINKED ISSUE #970: [BUG] Receiving notifications from logged out server
When we log out from a server, we still receive notifications from it.

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Closes #970

### CODE REVIEW

File: Rocket.Chat/API/Clients/PushClient.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    init(api: AnyAPIFetcher) {
+        self.api = api
+    }
+
+    func deletePushToken(token: String? = PushManager.getDeviceToken()) {
+        guard let token = token else { return }
+
+        api.fetch(PushTokenDeleteRequest(token: token), succeeded: nil, errored: { error in
+            if case .version = error {
+                Alert(key: "alert.push_token_error").present()
```
Comment (rafaelks) [inline Rocket.Chat/API/Clients/PushClient.swift]: Nice! :-)

File: Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+import SwiftyJSON
+
+typealias PushTokenDeleteResult = APIResult<PushTokenDeleteRequest>
+
+class PushTokenDeleteRequest: APIRequest {
+    let requiredVersion = Version(0, 60, 0)
+
+    let method = "DELETE"
```
Comment (rafaelks) [inline Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift]: I think we need a way to use constants here... maybe an ENUM? 🤔

File: Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    let path = "/api/v1/push.token"
```
Comment (rafaelks) [inline Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift]: Maybe constants here would be interesting too...

File: Rocket.Chat/Helpers/Alert.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        self.message = NSLocalizedString("\(key).message", comment: "")
+    }
+
+    func present() {
+        let window = UIWindow(frame: UIScreen.main.bounds)
+        window.rootViewController = UIViewController()
+        window.windowLevel = UIWindowLevelAlert + 1
+        window.makeKeyAndVisible()
+        window.rootViewController?.alert(title: title, message: message)
+    }
```
Comment (rafaelks) [inline Rocket.Chat/Helpers/Alert.swift]: Cool!!!

File: Rocket.Chat/Managers/DatabaseManager.swift [hunk 5]
```diff
@@ -160,6 +161,24 @@ struct DatabaseManager {
...
 
+    /**
+     This method gets the realm associated with this server
+    */
+    static func databaseInstace(index: Int) -> Realm? {
+        guard
+            let server = AuthManager.selectedServerInformation(index: index),
+            let databaseName = server[ServerPersistKeys.databaseName]
+        else {
+                return nil
```
Comment (rafaelks) [inline Rocket.Chat/Managers/DatabaseManager.swift]: <<<<

**Comment by rafaelks** [comment]
Made a few comments, do you want to adjust it or merge and adjust later?

File: Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift (hunk 3 above)
Comment (cardoso) [reply Rocket.Chat/API/Requests/General/PushTokenDeleteRequest.swift]: Some will be variables like rooms.upload/rid, so I'll think of a better solution later 🤔

**Comment by cardoso** [comment]
Alright @rafaelks, made the changes. 👍

File: Rocket.Chat/API/Requests/Command/RunCommandRequest.swift [hunk 6]
```diff
@@ -13,7 +13,7 @@ typealias RunCommandResult = APIResult<RunCommandRequest>
 typealias RunCommandSucceeded = (RunCommandResult) -> Void
 
 class RunCommandRequest: APIRequest {
-    let method = "POST"
+let method: HTTPMethod = .post
```
Comment (rafaelks) [inline Rocket.Chat/API/Requests/Command/RunCommandRequest.swift]: > >>>

Comment (cardoso) [reply Rocket.Chat/API/Requests/Command/RunCommandRequest.swift]: 😂 damn copy and paste, sorry. I wonder why hound is not reporting those 🤔

**Pull request merged**